<!DOCTYPE html>
<html>
<head>
  <title>ggtree for visualization and annotation of phylogenetic trees</title>
  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <link rel="stylesheet" href="index_files/rmdshower/shower/node_modules/shower-ribbon/styles/screen-4x3.css">
  <link rel="stylesheet" href="index_files/rmdshower/style-common.css">
  <link rel="stylesheet" href="index_files/rmdshower/style-ribbon.css">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.5.1/katex.min.css">
  <script src="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.5.1/katex.min.js"></script>
  <script src="index_files/rmdshower/auto-render.min.js" type="text/javascript"></script>
  
  
    <style type="text/css">
   div.sourceCode { overflow-x: auto; }
   table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
     margin: 0; padding: 0; vertical-align: baseline; border: none; }
   table.sourceCode { width: 100%; line-height: 100%; }
   td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
   td.sourceCode { padding-left: 5px; }
   code > span.kw { color: #0000ff; } /* Keyword */
   code > span.ch { color: #008080; } /* Char */
   code > span.st { color: #008080; } /* String */
   code > span.co { color: #008000; } /* Comment */
   code > span.ot { color: #ff4000; } /* Other */
   code > span.al { color: #ff0000; } /* Alert */
   code > span.er { color: #ff0000; font-weight: bold; } /* Error */
   code > span.wa { color: #008000; font-weight: bold; } /* Warning */
   code > span.cn { } /* Constant */
   code > span.sc { color: #008080; } /* SpecialChar */
   code > span.vs { color: #008080; } /* VerbatimString */
   code > span.ss { color: #008080; } /* SpecialString */
   code > span.im { } /* Import */
   code > span.va { } /* Variable */
   code > span.cf { color: #0000ff; } /* ControlFlow */
   code > span.op { } /* Operator */
   code > span.bu { } /* BuiltIn */
   code > span.ex { } /* Extension */
   code > span.pp { color: #ff4000; } /* Preprocessor */
   code > span.do { color: #008000; } /* Documentation */
   code > span.an { color: #008000; } /* Annotation */
   code > span.cv { color: #008000; } /* CommentVar */
   code > span.at { } /* Attribute */
   code > span.in { color: #008000; } /* Information */
  </style>
  
  
  
</head>

<body class="shower list">

  <header class="caption">
    <h1>ggtree for visualization and annotation of phylogenetic trees</h1>
    <p>YU Guangchuang</p>
  </header>

    
  
<section id="ggtree-for-visualization-and-annotation-of-phylogenetic-trees" class="slide level1">
<h1>ggtree for visualization and annotation of phylogenetic trees</h1>
<p><br></br></p>
<div style="position:absolute; bottom:200px;">
<h2 id="yu-guangchuang-">YU Guangchuang (余光创)</h2>
<p><a href="mailto:guangchuangyu@gmail.com">guangchuangyu@gmail.com</a></p>
</div>
<p><img src="figures/emoji2.png" class="cover"></p>
<p><img src="figures/hku-logo.jpg" width="220" style="position:absolute;bottom:140px;"></p>
<p><img src="figures/logo-chinar.png" width="200" style="position:absolute;bottom:80px;"></p>
</section>
<section id="what-is-a-phylogenetic-tree" class="slide level1">
<h1>What is a phylogenetic tree?</h1>
<h2 id="a-model-of-evolutionary-history">A model of evolutionary history</h2>
<ol type="1">
<li>Understanding life <strong>through time</strong> (over <strong>long periods of past time</strong>)</li>
<li>The connections between all groups of organisms as understood by <strong>ancestor-descendant</strong> relationships</li>
<li>A phylogenetic tree is <strong>one possible</strong> representation of the relationships</li>
<li>Evolutionary events like hybridization, horizontal gene transfer, recombination and reassortment can’t be modeled by a simple tree</li>
</ol>
<style>

.sourceCode {
    overflow: hidden;
}

pre {
    overflow: hidden;
}
</style>
<blockquote>
<p><font color='firebrick'>Models are approximations or simplications of real-world systems</font></p>
</blockquote>
<p><img src="figures/homer.png" width=300 style="position:absolute;right:25px;bottom:25px;"></p>
<p><link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.1/css/font-awesome.min.css"></p>
</section>
<section class="slide level1">

<h2 id="how-to-represent-phylogenetic-tree-in-computer-readable-form">How to represent phylogenetic tree in computer-readable form?</h2>
<h3 id="newick-format"><strong>Newick format</strong></h3>
<div style="float;">
<h3 id="developed-by-meacham-in-1984-in-felsensteins-phylip">developed by Meacham in 1984 in Felsenstein’s <em>PHYLIP</em></h3>
</div>
<div class="double">
<p class="double-flow">
<font color="#1B8DB3">((</font><font color="#478731">virus8</font><font color="firebrick">:0.05</font>,<font color="#1B8DB3">((</font><font color="#478731">virus6</font><font color="firebrick">:0.21</font>,<br> <font color="#1B8DB3">(((</font><font color="#478731">virus2</font><font color="firebrick">:0.39</font>,<font color="#478731">virus1</font><font color="#1B8DB3">)</font><font color="firebrick">:0.28</font><font color="#1B8DB3">)</font><font color="firebrick">:0.2</font>,<br> <font color="#478731">virus3</font><font color="#1B8DB3">)</font><font color="firebrick">:0.26</font><font color="#1B8DB3">)</font><font color="firebrick">:0.54</font>,<br> <font color="#1B8DB3">(</font><font color="#478731">virus5</font><font color="firebrick">:0.53</font>,<font color="#478731">virus4</font><font color="#1B8DB3">)))</font><font color="firebrick">:0.1</font><font color="#1B8DB3">)</font><font color="firebrick">:0.41</font><font color="#1B8DB3">)</font><font color="firebrick">:0.92</font><font color="#1B8DB3">)</font><font color="firebrick">:0.98</font>,<br> <font color="#478731">virus7</font><font color="#1B8DB3">))</font><font color="firebrick">:0.02</font>)<font color="firebrick">:0.08</font>)<font color="firebrick">:0.83</font>,<br> <font color="#1B8DB3">((</font><font color="#478731">virus10</font><font color="firebrick">:0.85</font>,<font color="#478731">virus9</font><font color="#1B8DB3">))</font><font color="firebrick">:0.5</font><font color="#1B8DB3">)</font><font color="firebrick">:0.98</font><font color="#1B8DB3">)</font>;
</p>
<p class="double-flow" style="position:absolute;right:20px;bottom:10px;">
<img src="index_files/figure-revealjs/unnamed-chunk-2-1.png" width="432" />
</p>
</div>
</section>
<section class="slide level1">

<div class="double">
<p class="double-flow">
<h3 id="phylogenetic-tree-file-formats"><strong>Phylogenetic tree file formats</strong></h3>
<p><font size="5"></p>
<ul>
<li>Newick</li>
<li>NEXUS</li>
<li>New Hampshire eXtended format (NHX)</li>
<li>NeXML</li>
<li>Phylip</li>
<li>jplace</li>
</ul>
<p></font></p>
<p><br></p>
</p>
<p class="double-flow">
<h3 id="output-files-from-various-software"><strong>Output files from various software</strong></h3>
<p><img src="figures/CdA0GboUMAE16PE.jpg" width="200" style="position:absolute;right:25px;bottom:25px;"> <font size="5"></p>
<ul>
<li>BEAST</li>
<li>EPA</li>
<li>HYPHY</li>
<li>PAML</li>
<li>PHYLODOG</li>
<li>pplacer</li>
<li>r8s</li>
<li>RAxML</li>
<li>RevBayes</li>
</ul>
<p></font></p>
</p>
</div>
</section>
<section class="slide level1">

<div style="position:absolute; top:60px;left:25px">
<h1 id="getting-trees-into-r-is-still-limited"><font color="firebrick">Getting trees into <code>R</code> is still <em><strong>limited</strong></em></h1>
</font>
</div>
<p><img src="figures/Screenshot%202016-04-08%2010.40.55.png" class="cover"></p>
<div style="position:absolute; bottom:25px;left:25px">
<p><a href="https://cran.r-project.org/web/views/Phylogenetics.html" class="uri">https://cran.r-project.org/web/views/Phylogenetics.html</a></p>
</div>
</section>
<section id="file-formats-supported-by-ggtree" class="slide level1">
<h1>File formats supported by <code>ggtree</code></h1>
<div class="double">
<p class="double-flow">
<h2 id="phylogenetic-tree-file-formats-1">Phylogenetic tree file formats</h2>
<ul>
<li>Newick (via <code>ape</code>)</li>
<li>NEXUS (via <code>ape</code>)</li>
<li>New Hampshire eXtended format (NHX)</li>
<li>Phylip</li>
<li>jplace</li>
</ul>
<p><br> <br></p>
</p>
<p class="double-flow">
<h2 id="output-files-from-various-software-1">Output files from various software</h2>
<ul>
<li>BEAST</li>
<li>EPA</li>
<li>HYPHY</li>
<li>PAML</li>
<li>PHYLODOG</li>
<li>pplacer</li>
<li>r8s</li>
<li>RAxML</li>
<li>RevBayes</li>
</ul>
</p>
</div>
</section>
<section id="s4-classes-defined-by-ggtree" class="slide level1">
<h1><em>S4</em> classes defined by <em>ggtree</em></h1>
<h2 id="to-store-phylogenetic-tree-with-evolutionary-evidences-inferred-by-different-software-packages">to store phylogenetic tree with evolutionary evidences (inferred by different software packages)</h2>
<div class="double">
<p class="double">
<ul>
<li>apeBootstrap</li>
<li>beast</li>
<li>codeml</li>
<li>codeml_mlc</li>
<li>hyphy</li>
</ul>
</p>
<p class="double-flow">
<ul>
<li>jplace</li>
<li>nhx</li>
<li>paml_rst</li>
<li>phangorn</li>
<li>r8s</li>
<li>raxml</li>
</ul>
</p>
</div>
<h2 id="section"><!--
----

<img src="figures/fig1.png" class="cover">
--></h2>
</section>
<section id="section-1" class="slide level1">
<h1></h1>
<h2 id="several-s3s4-classes-defined-in-other-r-packages-are-also-supported">Several <em>S3/S4</em> classes defined in other R packages are also supported</h2>
<h3 id="allow-integration-with-other-r-packages-to-analyze-and-visualize-phylogenies">allow integration with other R packages to analyze and visualize phylogenies</h3>
<ul>
<li>phylo (<code>ape</code>)</li>
<li>multiPhylo (<code>ape</code>)</li>
<li>phylo4 (<code>phylobase</code>)</li>
<li>phylo4d (<code>phylobase</code>)</li>
<li>obkData (<code>OutbreakTools</code>)</li>
<li>phyloseq (<code>phyloseq</code>)</li>
</ul>
</section>
<section class="slide level1">

<h2 id="ggtree-exploring-and-annotating-phylogenetic-trees">ggtree: exploring and annotating phylogenetic trees</h2>
<h2 id="section-2" class="fullpage"></h2>
<div class="fullpage width">
<p><img src="figures/Screenshot%202016-05-19%2014.27.22.png"></p>
</div>
</section>
<section class="slide level1">

<h2 id="merge_tree-allow-data-integration"><strong><em>merge_tree</em></strong>: allow data integration</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(ggtree)
beast_file &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;examples/MCC_FluA_H3.tree&quot;</span>, <span class="dt">package=</span><span class="st">&quot;ggtree&quot;</span>)
rst_file &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;examples/rst&quot;</span>, <span class="dt">package=</span><span class="st">&quot;ggtree&quot;</span>)
mlc_file &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;examples/mlc&quot;</span>, <span class="dt">package=</span><span class="st">&quot;ggtree&quot;</span>)

beast_tree &lt;-<span class="st"> </span><span class="kw">read.beast</span>(beast_file)
codeml_tree &lt;-<span class="st"> </span><span class="kw">read.codeml</span>(rst_file, mlc_file)
merged_tree &lt;-<span class="st"> </span><span class="kw">merge_tree</span>(beast_tree, codeml_tree)
<span class="kw">get.fields</span>(merged_tree)</code></pre></div>
<pre><code>##  [1] &quot;height&quot;           &quot;height_0.95_HPD&quot;  &quot;height_median&quot;   
##  [4] &quot;height_range&quot;     &quot;length&quot;           &quot;length_0.95_HPD&quot; 
##  [7] &quot;length_median&quot;    &quot;length_range&quot;     &quot;posterior&quot;       
## [10] &quot;rate&quot;             &quot;rate_0.95_HPD&quot;    &quot;rate_median&quot;     
## [13] &quot;rate_range&quot;       &quot;t&quot;                &quot;N&quot;               
## [16] &quot;S&quot;                &quot;dN_vs_dS&quot;         &quot;dN&quot;              
## [19] &quot;dS&quot;               &quot;N_x_dN&quot;           &quot;S_x_dS&quot;          
## [22] &quot;marginal_subs&quot;    &quot;joint_subs&quot;       &quot;marginal_AA_subs&quot;
## [25] &quot;joint_AA_subs&quot;</code></pre>
</section>
<section class="slide level1">

<h2 id="fortify-methods-for-converting-tree-objects-to-tidy-data"><code>fortify</code> methods for converting <em>tree objects</em> to <em>tidy data</em></h2>
<h3 id="data.frame-is-easy-to-manipulate-while-tree-object-is-hard"><code>data.frame</code> is easy to manipulate, while tree object is hard</h3>
<div style="position:absolute;left:100px;">
<p><font size='4'></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">fortify</span>(merged_tree) %&gt;%<span class="st"> </span>dplyr::<span class="kw">as_data_frame</span>()</code></pre></div>
<pre><code>## Source: local data frame [151 x 34]
## 
##     node parent branch.length         x     y                  label isTip
##    (int)  (int)         (dbl)     (dbl) (dbl)                  (chr) (lgl)
## 1      1    115     1.7132967 21.273403    42 A/Hokkaido/30-1-a/2013  TRUE
## 2      2     94     0.4987642 12.273403    19    A/New_York/334/2004  TRUE
## 3      3    149     1.1007898 13.273403    20    A/New_York/463/2005  TRUE
## 4      4     81     1.4515245  7.273403     5    A/New_York/452/1999  TRUE
## 5      5    150     0.7584041 13.273403    22    A/New_York/238/2005  TRUE
## 6      6     80     0.9862567  6.273403     4    A/New_York/523/1998  TRUE
## 7      7     85     1.0140833  8.273403     9    A/New_York/435/2000  TRUE
## 8      8     78     1.9788317  4.273403     2    A/New_York/582/1996  TRUE
## 9      9     79     1.1150139  4.273403     3    A/New_York/603/1996  TRUE
## 10    10     77     2.2734027  2.273403     1    A/New_York/657/1994  TRUE
## ..   ...    ...           ...       ...   ...                    ...   ...
## Variables not shown: branch (dbl), angle (dbl), height (dbl),
##   height_0.95_HPD (chr), height_median (dbl), height_range (chr), length
##   (dbl), length_0.95_HPD (chr), length_median (dbl), length_range (chr),
##   posterior (dbl), rate (dbl), rate_0.95_HPD (chr), rate_median (dbl),
##   rate_range (chr), t (dbl), N (dbl), S (dbl), dN_vs_dS (dbl), dN (dbl),
##   dS (dbl), N_x_dN (dbl), S_x_dS (dbl), marginal_subs (chr), joint_subs
##   (chr), marginal_AA_subs (chr), joint_AA_subs (chr)</code></pre>
<p></font></p>
</div>
<div style="position:absolute; bottom:10px;left:100px">
<font size="3"> <a href="http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeImport.html" class="uri">http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeImport.html</a> </font>
</div>
</section>
<section class="slide level1">

<h2 id="comparing-selection-pressure-codeml-with-substitution-rate-beast">Comparing selection pressure (<em>CodeML</em>) with substitution rate (<em>BEAST</em>)</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df &lt;-<span class="st"> </span><span class="kw">fortify</span>(merged_tree)
df &lt;-<span class="st"> </span>df[, <span class="kw">c</span>(<span class="st">&quot;dN_vs_dS&quot;</span>, <span class="st">&quot;dN&quot;</span>, <span class="st">&quot;dS&quot;</span>, <span class="st">&quot;rate&quot;</span>)]
df &lt;-<span class="st"> </span><span class="kw">na.omit</span>(df)
df &lt;-<span class="st"> </span>df[df$dN_vs_dS &gt;=<span class="dv">0</span> &amp;<span class="st"> </span>df$dN_vs_dS &lt;=<span class="fl">1.5</span>,] %&gt;%
<span class="st">         </span>tidyr::<span class="kw">gather</span>(type, value, dN_vs_dS:dS)
df$type[df$type ==<span class="st"> &quot;dN_vs_dS&quot;</span>] &lt;-<span class="st"> &quot;dN/dS&quot;</span>
df$type %&lt;&gt;%<span class="st"> </span><span class="kw">factor</span>(., <span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;dN/dS&quot;</span>, <span class="st">&quot;dN&quot;</span>, <span class="st">&quot;dS&quot;</span>))
<span class="kw">ggplot</span>(df, <span class="kw">aes</span>(rate, value)) +<span class="st"> </span><span class="kw">geom_hex</span>() +<span class="st"> </span>
<span class="st">    </span><span class="kw">facet_wrap</span>(~type, <span class="dt">scale=</span><span class="st">&quot;free_y&quot;</span>)</code></pre></div>
<p><img src="index_files/figure-revealjs/unnamed-chunk-6-1.png" width="768" /></p>
</section>
<section class="slide level1">

<div style="position:absolute; bottom:10px;left:80px">
<font size="4"> <a href="http://evolution.genetics.washington.edu/phylip/software.html#Plotting" class="uri">http://evolution.genetics.washington.edu/phylip/software.html#Plotting</a> </font>
</div>
<p><img src="figures/Screenshot%202016-04-15%2011.19.20.png" class="cover"></p>
</section>
<section class="slide level1">

<h2 id="why-another-tree-viewer" class="shout shrink">Why another tree viewer?</h2>
</section>
<section id="its-all-about-annotation" class="slide level1">
<h1>It’s all about <font color="firebrick"><strong><em>Annotation</em></strong></font></h1>
<p><img src="figures/codeml.png" width="400" style="position:absolute;right:0px;bottom:400px;"> <img src="figures/figtree2.png" width="380" style="position:absolute;right:8px;bottom:83px;"></p>
<ul>
<li>Traditional tree visualization software
<ul>
<li>modifying node labels</li>
</ul></li>
<li>A subset of sofware allows users to annotate trees
<ul>
<li><em>FigTree</em>, <em>TreeDyn</em>, <em>iTOL etc</em></li>
<li>pre-defined annotating utilities
<ul>
<li>coloring branches, highlighting clades <em>etc</em></li>
</ul></li>
<li>focus on specific analyses and data types</li>
</ul></li>
</ul>
</section>
<section class="slide level1">

<h2 id="plotting-trees-data">PLOTTING TREES + DATA</h2>
<blockquote>
<p>I’m often asked how we make nice annotated tree figures for our papers. The real answer is “with creativity, practice and a lot of trial and error”. But of course the answer people are looking for is what tools do we use.</p>
<p>There are LOADS of different tools out there for plotting trees in various ways. <font color='firebrick'>I’ve listed my favourites for basic tree viewing below, but these really only work for quite simple visualisations, have differing and often non-overlapping feature sets, and are hard to customise.</font> My solution to such problems is generally to turn to Python + R, and that is exactly the case here.</p>
</blockquote>
<p><font size="4">Kart Holt, <a href="https://holtlab.net/2015/10/03/plotting-trees-data/" class="uri">https://holtlab.net/2015/10/03/plotting-trees-data/</a></font></p>
<p><font size="4">Tools listed: <em>Dendroscrope</em>, <em>FigTree</em>, <em>MicroReact</em>, <em>Interactive tree of life (iTOL)</em> and <em>ScripTree</em>.</font></p>
</section>
<section id="programmable-software-wanted" class="slide level1">
<h1><font color="firebrick"><strong>Programmable</strong></font> software wanted</h1>
<ul>
<li>incorporate various types and sources of data</li>
<li>freely combine different layers of annotation</li>
<li>allow high levels of customization</li>
<li>reproducibility, automation &amp; communication</li>
</ul>
<p><img src="figures/Screenshot%202016-04-20%2016.29.54.png" width="700" style="position:absolute;bottom:25px;"></p>
<h2 id="section-3"><!--
----

## Why visualizing phylogenetic trees using `ggplot2`?

> 1. Follows a <font color="steelblue">grammar</font> (of graphics), just like any language (Wilkinson, 2005)
> 2. A <font color="steelblue">plot</font> is made up of multiple layers
> 3. A <font color="steelblue">layer</font> consists of data, a set of <font color="steelblue">aes</font>thetics mappings, a <font color="steelblue">geom</font>emtric object and a <font color="steelblue">stat</font>istical transformation
> 4. Layers are independent and reusable
> 4. High level of customization (the plots can be edited after they are ready)
> 5. `R` users are already familiar with `ggplot2` syntax

--></h2>
</section>
<section id="the-ggtree-package" class="slide level1">
<h1>The <code>ggtree</code> package</h1>
<ul>
<li>parsing tree data from a various file formats</li>
<li>extends <code>ggplot2</code> to support tree objects</li>
<li>geometric layers for tree annotation</li>
<li>helper functions for tree manipulation</li>
<li>complex tree annotation</li>
</ul>
<img src="figures/fcimb-06-00036-g003.jpg" height="250" style="position:absolute;right:20px;bottom:320px;">
<div style="position:absolute; bottom:280px;right:20px">
<font size="3"><a href="http://doi.org/10.3389/fcimb.2016.00036" class="uri">http://doi.org/10.3389/fcimb.2016.00036</a> </font>
</div>
<p><img src="figures/Screenshot%202016-04-19%2019.08.41.png" height="130" style="position:absolute;left:0px;bottom:50px;"> <img src="figures/Screenshot%202016-04-19%2019.30.47.png" height="130" style="position:absolute;left:335px;bottom:50px;"> <img src="figures/Screenshot%202016-04-19%2019.26.23.png" height="130" style="position:absolute;right:0px;bottom:50px;"></p>
<div style="position:absolute; bottom:15px;right:20px">
<font size="3"> <i class="fa fa-weibo fa-lg"></i> <a href="http://huati.weibo.com/k/ggtree" class="uri">http://huati.weibo.com/k/ggtree</a> <i class="fa fa-twitter fa-lg"></i> <a href="https://twitter.com/hashtag/ggtree" class="uri">https://twitter.com/hashtag/ggtree</a> </font>
</div>
</section>
<section class="slide level1">

<h2 id="other-ggplot2-based-tree-viewers">Other <code>ggplot2</code>-based tree viewers</h2>
<ul>
<li><a href="https://cran.r-project.org/web/packages/OutbreakTools/index.html">OutbreakTools</a> (<a href="http://www.ncbi.nlm.nih.gov/pubmed/24928667">2014 Epidemics</a>)</li>
<li><a href="https://bioconductor.org/packages/phyloseq/">phyloseq</a> (<a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0061217">2013 PLOS One</a>)</li>
<li>designed for specific need and data
<ul>
<li>not applicable for widely used tree file formats</li>
<li>plot functions (<code>plotggphy</code>, <code>plot_tree</code>) based on <code>ggplot2</code></li>
<li>provide limited control (parameters) over the graphic output</li>
</ul></li>
<li><strong><em>grammar of graphics</em></strong> is missing
<ul>
<li>no layers provided for tree annotation</li>
<li>parse tree as a collection of line segments</li>
<li>adding extra layer of tree annotation is quite challenging</li>
</ul></li>
</ul>
</section>
<section class="slide level1">

<h2 id="the-ggtree-package-1">The <code>ggtree</code> package</h2>
<ul>
<li>motivation: high level of annotation using grammar of graphics</li>
<li>(for visualization) parse tree as a collection of nodes</li>
<li>extend <code>ggplot2</code>
<ul>
<li>extend <code>ggplot</code> function to support tree objects</li>
<li><code>geom_tree</code>: extends <code>ggplot2</code> to support tree structure</li>
<li>geom layers for tree annotation</li>
</ul></li>
</ul>
</section>
<section class="slide level1">

<h2 id="geom_tree-geometric-layer-to-support-tree-structure"><code>geom_tree</code>: geometric layer to support tree structure</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(beast_tree, <span class="kw">aes</span>(x,y)) +<span class="st"> </span><span class="kw">geom_tree</span>() </code></pre></div>
<img src="index_files/figure-revealjs/unnamed-chunk-8-1.png" width="768" />
<div style="position:absolute; bottom:10px;left:120px">
<font size="3"> <a href="http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeVisualization.html" class="uri">http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeVisualization.html</a> </font>
</div>
</section>
<section class="slide level1">

<h2 id="theme_tree-theme_tree2"><code>theme_tree</code> &amp; <code>theme_tree2</code></h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">multiplot</span>(p +<span class="st"> </span><span class="kw">theme_tree</span>(), p +<span class="st"> </span><span class="kw">theme_tree2</span>(), <span class="dt">ncol=</span><span class="dv">2</span>)</code></pre></div>
<p><img src="index_files/figure-revealjs/unnamed-chunk-9-1.png" width="768" /></p>
<div style="position:absolute; bottom:10px;left:120px">
<font size="3"> <a href="http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeVisualization.html" class="uri">http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeVisualization.html</a> </font>
</div>
</section>
<section class="slide level1">

<h2 id="ggtree-ggplot-geom_tree-theme_tree"><code>ggtree</code> = <code>ggplot()</code> + <code>geom_tree()</code> + <code>theme_tree()</code></h2>
<p><font size="5"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(beast_tree)</code></pre></div>
<p><img src="index_files/figure-revealjs/unnamed-chunk-10-1.png" width="576" /> </font></p>
<div style="position:absolute; bottom:10px;left:120px">
<font size="3"> <a href="http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeVisualization.html" class="uri">http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeVisualization.html</a> </font>
</div>
</section>
<section class="slide level1">

<h2 id="ggtree-ggplot-geom_tree-theme_tree-1"><code>ggtree</code> = <code>ggplot()</code> + <code>geom_tree()</code> + <code>theme_tree()</code></h2>
<p><font size="5"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(beast_tree, <span class="dt">branch.length=</span><span class="st">&#39;none&#39;</span>, <span class="dt">color=</span><span class="st">&#39;slateblue&#39;</span>,
       <span class="dt">layout=</span><span class="st">&#39;fan&#39;</span>, <span class="dt">open.angle=</span><span class="dv">180</span>, <span class="dt">size=</span><span class="dv">2</span>)</code></pre></div>
<p><img src="index_files/figure-revealjs/unnamed-chunk-11-1.png" width="576" /> </font></p>
<div style="position:absolute; bottom:10px;left:120px">
<font size="3"> <a href="http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeVisualization.html" class="uri">http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeVisualization.html</a> </font>
</div>
</section>
<section class="slide level1">

<h2 id="ggtree-ggplot-geom_tree-theme_tree-2"><code>ggtree</code> = <code>ggplot()</code> + <code>geom_tree()</code> + <code>theme_tree()</code></h2>
<p><font size="5"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(beast_tree, <span class="kw">aes</span>(<span class="dt">color=</span>rate), <span class="dt">linetype=</span><span class="st">&#39;dashed&#39;</span>)</code></pre></div>
<p><img src="index_files/figure-revealjs/unnamed-chunk-12-1.png" width="576" /> </font></p>
<div style="position:absolute; bottom:10px;left:120px">
<font size="3"> <a href="http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeVisualization.html" class="uri">http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeVisualization.html</a> </font>
</div>
</section>
<section class="slide level1">

<h2 id="layers-defined-in-ggplot2-can-be-used-directly-in-ggtree">layers defined in <code>ggplot2</code> can be used directly in <code>ggtree</code></h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p &lt;-<span class="st"> </span><span class="kw">ggtree</span>(beast_tree)
(p +<span class="st"> </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">color=</span>isTip), <span class="dt">size=</span><span class="dv">1</span>) +<span class="st"> </span>
<span class="st">    </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">label=</span><span class="kw">round</span>(posterior*<span class="dv">100</span>)),
        <span class="dt">hjust=</span>-.<span class="dv">1</span>, <span class="dt">size=</span><span class="fl">2.5</span>)) %&gt;%<span class="st"> </span><span class="kw">viewClade</span>(<span class="dt">node=</span><span class="dv">118</span>)   </code></pre></div>
<p><img src="index_files/figure-revealjs/unnamed-chunk-13-1.png" width="576" /></p>
</section>
<section class="slide level1">

<h2 id="add-information-to-selected-nodes">Add information to selected node(s)</h2>
<ul>
<li><code>ggplo2</code> doesn’t support <font color="firebrick"><strong><em>subsetting</em></strong></font></li>
<li><code>ggtree</code> provides modified version of geometric layers to support it
<ul>
<li>geom_label2</li>
<li>geom_text2</li>
<li>geom_point2</li>
<li>geom_segment2</li>
</ul></li>
</ul>
</section>
<section class="slide level1">

<h2 id="modified-version-of-geometric-layers-to-support-subsetting">modified version of geometric layers to support <font color="firebrick"><strong><em>subsetting</em></strong></font></h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">(p +<span class="st"> </span><span class="kw">geom_point2</span>(<span class="kw">aes</span>(<span class="dt">subset=</span> !<span class="kw">is.na</span>(posterior) &amp;<span class="st"> </span>posterior &gt;<span class="st"> </span>.<span class="dv">75</span>), 
                <span class="dt">color=</span><span class="st">&#39;firebrick&#39;</span>)) %&gt;%<span class="st"> </span><span class="kw">viewClade</span>(<span class="dt">node=</span><span class="dv">118</span>)</code></pre></div>
<p><img src="index_files/figure-revealjs/unnamed-chunk-14-1.png" width="768" /></p>
</section>
<section class="slide level1">

<h2 id="geometric-layers-for-phylogenetic-tree-annotation">geometric layers for phylogenetic tree annotation</h2>
<ul>
<li>geom_treescale</li>
<li>geom_tiplab</li>
<li>geom_hilight</li>
<li>geom_cladelabel</li>
<li>geom_range</li>
<li>geom_taxalink</li>
</ul>
<p><img src="figures/ggtree_layers.png" width="650" style="position:absolute;right:10px;bottom:40px;"></p>
<div style="position:absolute; bottom:10px;left:100px">
<font size="3"> <a href="http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeAnnotation.html" class="uri">http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeAnnotation.html</a> </font>
</div>
</section>
<section class="slide level1">

<h2 id="geom_treescale-displays-the-tree-scale"><code>geom_treescale</code> displays the tree scale</h2>
<h3 id="scale-of-branch-length-e.g.substitution-rate-divergent-time">scale of branch length, e.g. substitution rate, divergent time</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p &lt;-<span class="st"> </span>p +<span class="st"> </span><span class="kw">geom_treescale</span>(<span class="dt">x=</span><span class="dv">0</span>, <span class="dt">y=</span><span class="dv">10</span>, <span class="dt">offset=</span><span class="dv">2</span>)</code></pre></div>
<p><img src="index_files/figure-revealjs/unnamed-chunk-16-1.png" width="768" /></p>
</section>
<section class="slide level1">

<h2 id="geom_tiplab-layer-of-tip-labels-taxa-names"><code>geom_tiplab</code>: layer of tip labels (taxa names)</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p2 &lt;-<span class="st"> </span>p +<span class="st"> </span><span class="kw">geom_tiplab</span>(<span class="dt">size=</span><span class="dv">2</span>)
p &lt;-<span class="st"> </span>p +<span class="st"> </span><span class="kw">geom_tiplab</span>(<span class="dt">align=</span>T, <span class="dt">size=</span><span class="dv">2</span>, <span class="dt">linesize=</span>.<span class="dv">3</span>) +<span class="st"> </span><span class="kw">xlim</span>(<span class="ot">NA</span>, <span class="dv">30</span>)
<span class="kw">multiplot</span>(p2, p, <span class="dt">ncol=</span><span class="dv">2</span>)</code></pre></div>
<p><img src="index_files/figure-revealjs/unnamed-chunk-17-1.png" width="768" /> </font></p>
</section>
<section class="slide level1">

<h2 id="geom_hilight-highlight-clade-with-background-color"><code>geom_hilight</code>: highlight clade with background color</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p &lt;-<span class="st"> </span>p +<span class="st"> </span><span class="kw">geom_hilight</span>(<span class="dt">node=</span><span class="dv">118</span>, <span class="dt">fill=</span><span class="st">&#39;blue&#39;</span>, <span class="dt">alpha=</span>.<span class="dv">4</span>)</code></pre></div>
<p><img src="index_files/figure-revealjs/unnamed-chunk-19-1.png" width="768" /></p>
</section>
<section class="slide level1">

<h2 id="geom_cladelabel-labeling-clade"><code>geom_cladelabel</code>: labeling clade</h2>
<p><font size="4"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p &lt;-<span class="st"> </span>p +<span class="st"> </span><span class="kw">geom_cladelabel</span>(<span class="dv">118</span>, <span class="st">&#39;a selected clade&#39;</span>, <span class="dt">offset=</span><span class="fl">6.5</span>, <span class="dt">offset.text=</span>.<span class="dv">5</span>, 
                         <span class="dt">barsiz=</span><span class="dv">3</span>, <span class="dt">color=</span><span class="st">&#39;darkgreen&#39;</span>, <span class="dt">angle=</span><span class="dv">90</span>, <span class="dt">hjust=</span><span class="st">&#39;center&#39;</span>)</code></pre></div>
<p></font></p>
<p><img src="index_files/figure-revealjs/unnamed-chunk-21-1.png" width="768" /></p>
</section>
<section class="slide level1">

<h2 id="geom_range-displays-uncertainty-of-branch-lengths"><code>geom_range</code>: displays uncertainty of branch lengths</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p &lt;-<span class="st"> </span>p +<span class="st"> </span><span class="kw">geom_range</span>(<span class="st">&#39;height_0.95_HPD&#39;</span>, <span class="dt">color=</span><span class="st">&#39;red&#39;</span>, <span class="dt">alpha=</span>.<span class="dv">5</span>)</code></pre></div>
<p><img src="index_files/figure-revealjs/unnamed-chunk-23-1.png" width="768" /></p>
</section>
<section class="slide level1">

<h2 id="geom_taxalink-connecting-related-taxa"><code>geom_taxalink</code>: connecting related taxa</h2>
<p><font size="4"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p &lt;-<span class="st"> </span>p +<span class="st"> </span><span class="kw">geom_taxalink</span>(<span class="dv">45</span>, <span class="dv">19</span>, <span class="dt">color=</span><span class="st">&#39;purple&#39;</span>, <span class="dt">arrow=</span><span class="kw">arrow</span>(<span class="dt">length=</span><span class="kw">unit</span>(<span class="fl">0.02</span>, <span class="st">&#39;npc&#39;</span>))) +<span class="st"> </span>
<span class="st">         </span><span class="kw">geom_taxalink</span>(<span class="dv">46</span>, <span class="dv">18</span>, <span class="dt">color=</span><span class="st">&#39;blue&#39;</span>, <span class="dt">arrow=</span><span class="kw">arrow</span>(<span class="dt">length=</span><span class="kw">unit</span>(<span class="fl">0.02</span>, <span class="st">&#39;npc&#39;</span>)))</code></pre></div>
<p></font></p>
<p><img src="index_files/figure-revealjs/unnamed-chunk-25-1.png" width="768" /></p>
</section>
<section class="slide level1">

<h2 id="annotate-with-users-own-data-from-external-files">Annotate with user’s own data (from external files)</h2>
<p><img src="figures/tree_dot.png" width="520" style="position:absolute;right:10px;bottom:50px;"></p>
<div style="float:left;">
<p><font size="5"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## p  + geom_tippoint(aes(color=grepl(&quot;Swine&quot;, label)))

tip &lt;-<span class="st"> </span><span class="kw">get.tree</span>(beast_tree)$tip.label
host &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="st">&quot;Human&quot;</span>, <span class="kw">length</span>(tip))
host[<span class="kw">grep</span>(<span class="st">&quot;Swine&quot;</span>, tip)] &lt;-<span class="st"> &quot;Swine&quot;</span>
host.df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">taxa=</span>tip, <span class="dt">host=</span>host)
p %&lt;+%<span class="st"> </span>host.df +<span class="st"> </span><span class="kw">geom_tippoint</span>(<span class="kw">aes</span>(<span class="dt">color=</span>host))</code></pre></div>
<p></font></p>
</div>
<div style="position:absolute; bottom:10px;left:100px">
<font size="3"> <a href="http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeAnnotation.html" class="uri">http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeAnnotation.html</a> </font>
</div>
</section>
<section class="slide level1">

<h2 id="tree-annotation-example">tree annotation example</h2>
<p><img src="figures/tree_dot.png" width="520" style="position:absolute;right:10px;bottom:50px;"></p>
<p><font size='4'></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(ggtree)
beast_file &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;examples/MCC_FluA_H3.tree&quot;</span>, <span class="dt">package=</span><span class="st">&quot;ggtree&quot;</span>)
beast_tree &lt;-<span class="st"> </span><span class="kw">read.beast</span>(beast_file)

<span class="kw">ggtree</span>(beast_tree) +<span class="st"> </span><span class="kw">geom_treescale</span>(<span class="dt">x=</span><span class="dv">0</span>, <span class="dt">y=</span><span class="dv">10</span>, <span class="dt">offset=</span><span class="dv">2</span>) +<span class="st"> </span>
<span class="st">    </span><span class="kw">geom_tiplab</span>(<span class="dt">align=</span>T, <span class="dt">size=</span><span class="dv">2</span>, <span class="dt">linesize=</span>.<span class="dv">3</span>) +<span class="st"> </span><span class="kw">xlim</span>(<span class="ot">NA</span>, <span class="dv">30</span>) +<span class="st"> </span>
<span class="st">    </span><span class="kw">geom_hilight</span>(<span class="dt">node=</span><span class="dv">118</span>, <span class="dt">fill=</span><span class="st">&#39;blue&#39;</span>, <span class="dt">alpha=</span>.<span class="dv">4</span>) +<span class="st"> </span>
<span class="st">    </span><span class="kw">geom_cladelabel</span>(<span class="dv">118</span>, <span class="st">&#39;a selected clade&#39;</span>, <span class="dt">offset=</span><span class="fl">6.5</span>,
                    <span class="dt">offset.text=</span>.<span class="dv">5</span>, <span class="dt">barsiz=</span><span class="dv">3</span>, <span class="dt">color=</span><span class="st">&#39;darkgreen&#39;</span>,
                    <span class="dt">angle=</span><span class="dv">90</span>, <span class="dt">hjust=</span><span class="st">&#39;center&#39;</span>) +
<span class="st">    </span><span class="kw">geom_range</span>(<span class="st">&#39;height_0.95_HPD&#39;</span>, <span class="dt">color=</span><span class="st">&#39;red&#39;</span>, <span class="dt">alpha=</span>.<span class="dv">5</span>) +
<span class="st">    </span><span class="kw">geom_taxalink</span>(<span class="dv">45</span>, <span class="dv">19</span>, <span class="dt">color=</span><span class="st">&#39;purple&#39;</span>,
                  <span class="dt">arrow=</span><span class="kw">arrow</span>(<span class="dt">length=</span><span class="kw">unit</span>(<span class="fl">0.02</span>, <span class="st">&#39;npc&#39;</span>))) +
<span class="st">    </span><span class="kw">geom_taxalink</span>(<span class="dv">46</span>, <span class="dv">18</span>, <span class="dt">color=</span><span class="st">&#39;blue&#39;</span>,
                  <span class="dt">arrow=</span><span class="kw">arrow</span>(<span class="dt">length=</span><span class="kw">unit</span>(<span class="fl">0.02</span>, <span class="st">&#39;npc&#39;</span>))) +
<span class="st">    </span><span class="kw">geom_tippoint</span>(<span class="kw">aes</span>(<span class="dt">color=</span><span class="kw">grepl</span>(<span class="st">&quot;Swine&quot;</span>, label)))</code></pre></div>
<p></font></p>
<div style="position:absolute; bottom:10px;left:100px">
<font size="3"> <a href="http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeAnnotation.html" class="uri">http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeAnnotation.html</a> </font>
</div>
</section>
<section id="tree-manipulation" class="slide level1">
<h1>Tree manipulation</h1>
<ul>
<li><font color='firebrick'><strong>groupClade &amp; groupOTU</strong></font></li>
<li><font color='grey'>scaleClade</font></li>
<li><font color='grey'>rotate &amp; flip</font></li>
<li><font color='grey'>collapse &amp; expand</font></li>
<li><font color='grey'>open_tree &amp; rotate_tree</font></li>
<li><font color='grey'>rescale_tree</font></li>
</ul>
<img src="figures/groupOTU1.png" height="300" style="position:absolute;right:50px;bottom:320px;"> <img src="figures/groupOTU2.png" height="300" style="position:absolute;right:50px;bottom:30px;">
<div style="position:absolute; bottom:10px;left:80px">
<font size="3"> <a href="http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeManipulation.html" class="uri">http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeManipulation.html</a> </font>
</div>
</section>
<section id="tree-manipulation-1" class="slide level1">
<h1>Tree manipulation</h1>
<ul>
<li><font color='grey'>groupClade &amp; groupOTU</font></li>
<li><font color='firebrick'><strong>scaleClade</strong></font></li>
<li><font color='grey'>rotate &amp; flip</font></li>
<li><font color='grey'>collapse &amp; expand</font></li>
<li><font color='grey'>open_tree &amp; rotate_tree</font></li>
<li><font color='grey'>rescale_tree</font></li>
</ul>
<img src="figures/scaleClade.png" height="300" style="position:absolute;right:50px;bottom:50px;">
<div style="position:absolute; bottom:10px;left:80px">
<font size="3"> <a href="http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeManipulation.html" class="uri">http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeManipulation.html</a> </font>
</div>
</section>
<section id="tree-manipulation-2" class="slide level1">
<h1>Tree manipulation</h1>
<ul>
<li><font color='grey'>groupClade &amp; groupOTU</font></li>
<li><font color='grey'>scaleClade</font></li>
<li><font color='firebrick'><strong>rotate &amp; flip</strong></font></li>
<li><font color='grey'>collapse &amp; expand</font></li>
<li><font color='grey'>open_tree &amp; rotate_tree</font></li>
<li><font color='grey'>rescale_tree</font></li>
</ul>
<p><img src="figures/flip.png" height="250" style="position:absolute;right:50px;bottom:50px;"></p>
<img src="figures/rotate_clade.gif" width="380" style="position:absolute;right:50px;bottom:300px;">
<div style="position:absolute; bottom:10px;left:80px">
<font size="3"> <a href="http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeManipulation.html" class="uri">http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeManipulation.html</a> </font>
</div>
</section>
<section id="tree-manipulation-3" class="slide level1">
<h1>Tree manipulation</h1>
<ul>
<li><font color='grey'>groupClade &amp; groupOTU</font></li>
<li><font color='grey'>scaleClade</font></li>
<li><font color='grey'>rotate &amp; flip</font></li>
<li><font color='firebrick'><strong>collapse &amp; expand</strong></font></li>
<li><font color='grey'>open_tree &amp; rotate_tree</font></li>
<li><font color='grey'>rescale_tree</font></li>
</ul>
<p><img src="figures/collapse_expand.png" height="200" style="position:absolute;right:20px;bottom:50px;"></p>
<div style="position:absolute; bottom:10px;left:80px">
<font size="3"> <a href="http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeManipulation.html" class="uri">http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeManipulation.html</a> </font>
</div>
</section>
<section id="tree-manipulation-4" class="slide level1">
<h1>Tree manipulation</h1>
<ul>
<li><font color='grey'>groupClade &amp; groupOTU</font></li>
<li><font color='grey'>scaleClade</font></li>
<li><font color='grey'>rotate &amp; flip</font></li>
<li><font color='grey'>collapse &amp; expand</font></li>
<li><font color='firebrick'><strong>open_tree &amp; rotate_tree</strong></font></li>
<li><font color='grey'>rescale_tree</font></li>
</ul>
<p><img src="figures/fan_layout.gif" width="400" style="position:absolute;right:20px;bottom:330px;"> <img src="figures/rotate_tree.gif" width="400" style="position:absolute;right:50px;bottom:0px;"></p>
<div style="position:absolute; bottom:10px;left:80px">
<font size="3"> <a href="http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeManipulation.html" class="uri">http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeManipulation.html</a> </font>
</div>
</section>
<section id="tree-manipulation-5" class="slide level1">
<h1>Tree manipulation</h1>
<ul>
<li><font color='grey'>groupClade &amp; groupOTU</font></li>
<li><font color='grey'>scaleClade</font></li>
<li><font color='grey'>rotate &amp; flip</font></li>
<li><font color='grey'>collapse &amp; expand</font></li>
<li><font color='grey'>open_tree &amp; rotate_tree</font></li>
<li><font color='firebrick'><strong>rescale_tree</strong></font></li>
</ul>
<!--
<img src="figures/codeml.png" width="580" style="position:absolute;right:20px;bottom:320px;">
-->
<div style="position:absolute; top:180px;right:20px">
<p><font size="4"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p1 &lt;-<span class="st"> </span><span class="kw">ggtree</span>(merged_tree, <span class="kw">aes</span>(<span class="dt">color=</span>rate)) +<span class="st"> </span>
<span class="st">         </span><span class="kw">theme_tree2</span>() +<span class="st"> </span><span class="kw">geom_tiplab</span>(<span class="dt">size=</span><span class="fl">1.5</span>) +<span class="st"> </span>
<span class="st">           </span><span class="kw">scale_color_continuous</span>(<span class="dt">high=</span><span class="st">&#39;red&#39;</span>, <span class="dt">low=</span><span class="st">&#39;green&#39;</span>)

p2 &lt;-<span class="st"> </span>p1 %&lt;%<span class="st"> </span><span class="kw">rescale_tree</span>(merged_tree, <span class="st">&#39;dN&#39;</span>) +<span class="st"> </span><span class="kw">aes</span>(<span class="dt">color=</span>dN)
<span class="kw">multiplot</span>(p1+<span class="kw">xlim</span>(<span class="ot">NA</span>, <span class="dv">25</span>)+<span class="kw">ggtitle</span>(<span class="st">&#39;time-scaled tree&#39;</span>), 
    p2+<span class="kw">xlim</span>(<span class="ot">NA</span>, <span class="fl">0.07</span>)+<span class="kw">ggtitle</span>(<span class="st">&#39;dN tree&#39;</span>), <span class="dt">ncol=</span><span class="dv">2</span>)</code></pre></div>
<p></font></p>
</div>
<p><img src="figures/rescale_tree.png" width="580" style="position:absolute;right:50px;top:380px;"></p>
<div style="position:absolute; bottom:10px;left:80px">
<font size="3"> <a href="http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeManipulation.html" class="uri">http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeManipulation.html</a> </font>
</div>
<!--

----

## Tree manipulation: *rotating* a clade

<font size="5">

```r
set.seed(2016-05-29)
p <- ggtree(tree <- rtree(50)) + geom_tiplab()
for (n in reorder(tree, 'postorder')$edge[,1] %>% unique) {
    p <- rotate(p, n)
    print(p + geom_point2(aes(subset = (node == n)), color='red'))
    animation::ani.pause(0.3)
}
```
</font>


<img src="figures/rotate_clade.gif" width="370" style="position:absolute;left:80px;bottom:30px;">
<div style='position:absolute; bottom:10px;left:80px'>
<font size="3">
<http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeManipulation.html>
</font></div>



----

## Tree manipulation: *rescale* branch lengths
<font size="5">

```r
p1 <- ggtree(merged_tree, aes(color=rate)) + 
         theme_tree2() + geom_tiplab(size=1.5) + 
           scale_color_continuous(high='red', low='green')
p2 <- p1 %<% rescale_tree(merged_tree, 'dN') + aes(color=dN)
multiplot(p1+xlim(NA, 25)+ggtitle('time-scaled tree'), 
      p2+xlim(NA, 0.07)+ggtitle('dN tree'), ncol=2)
```

<img src="index_files/figure-revealjs/unnamed-chunk-29-1.png" width="768" />
</font>
<div style='position:absolute; bottom:10px;left:80px'>
<font size="3">
<http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeManipulation.html>
</font></div>
-->
</section>
<section class="slide level1">

<h2 id="more-complex-tree-annotation">More complex tree annotation</h2>
<ul>
<li><font color='firebrick'><strong>associated matrix</strong></font></li>
<li><font color='grey'>multiple sequence alignment</font></li>
<li><font color='grey'>insets/subplots</font></li>
<li><font color='grey'>phylopic &amp; local images</font></li>
</ul>
<p><img src="figures/heatmap.png" height="600" style="position:absolute;right:30px;bottom:50px;"></p>
<div style="position:absolute; bottom:10px;left:80px">
<font size="3"> <a href="http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/advanceTreeAnnotation.html" class="uri">http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/advanceTreeAnnotation.html</a> </font>
</div>
</section>
<section class="slide level1">

<h2 id="more-complex-tree-annotation-1">More complex tree annotation</h2>
<ul>
<li><font color='grey'>associated matrix</font></li>
<li><font color='firebrick'><strong>multiple sequence alignment</strong></font></li>
<li><font color='grey'>insets/subplots</font></li>
<li><font color='grey'>phylopic &amp; local images</font></li>
</ul>
<p><img src="figures/msa.png" height="380" style="position:absolute;right:30px;bottom:30px;"></p>
<div style="position:absolute; bottom:10px;left:80px">
<font size="3"> <a href="http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/advanceTreeAnnotation.html" class="uri">http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/advanceTreeAnnotation.html</a> </font>
</div>
</section>
<section class="slide level1">

<h2 id="more-complex-tree-annotation-2">More complex tree annotation</h2>
<ul>
<li><font color='grey'>associated matrix</font></li>
<li><font color='grey'>multiple sequence alignment</font></li>
<li><font color='firebrick'><strong>insets/subplots</strong></font></li>
<li><font color='grey'>phylopic &amp; local images</font></li>
</ul>
<p><img src="figures/inset.png" height="420" style="position:absolute;right:30px;bottom:80px;"></p>
<div style="position:absolute; bottom:10px;left:80px">
<font size="3"> <a href="http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/advanceTreeAnnotation.html" class="uri">http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/advanceTreeAnnotation.html</a> </font>
</div>
</section>
<section class="slide level1">

<h2 id="more-complex-tree-annotation-3">More complex tree annotation</h2>
<ul>
<li><font color='grey'>associated matrix</font></li>
<li><font color='grey'>multiple sequence alignment</font></li>
<li><font color='grey'>insets/subplots</font></li>
<li><font color='firebrick'><strong>phylopic &amp; local images</strong></font></li>
</ul>
<p><img src="figures/2016-05-05-145429_731x430_scrot.png" width="400" style="position:absolute;right:30px;bottom:450px;"></p>
<p><img src="figures/phylopic.png" width="400" style="position:absolute;right:30px;bottom:30px;"></p>
<p><img src="figures/2015_peerJ_fig-9-2x.jpg" width="250" style="position:absolute;left:100px;bottom:55px;"></p>
<div style="position:absolute; bottom:40px;left:80px">
<font size="3"> <a href="https://doi.org/10.7717/peerj.976" class="uri">https://doi.org/10.7717/peerj.976</a> </font>
</div>
<div style="position:absolute; bottom:10px;left:80px">
<font size="3"> <a href="http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/advanceTreeAnnotation.html" class="uri">http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/advanceTreeAnnotation.html</a> </font>
</div>
</section>
<section class="slide level1">

<h2 id="add-values-to-ggplot2">add values to <code>ggplot2</code></h2>
<ul>
<li>support tree structure</li>
<li>support subsetting</li>
<li>support adding subplots</li>
</ul>
<p><img src="figures/Screenshot%202016-04-20%2019.41.47.png" width="520" style="position:absolute;right:30px;bottom:60px;"></p>
<p><img src="figures/Screenshot%202016-04-20%2019.42.20.png" width="400" style="position:absolute;left:80px;bottom:60px;"></p>
</section>
<section class="slide level1">

<div style="position:absolute;top:50px;">
<h1 id="conclusions">Conclusions</h1>
<h2 id="getting-tree-into-r">Getting tree into <code>R</code></h2>
<p><font size='5'></p>
<ul>
<li>tree parsers: bring evolution evidences to be used/analyzed in <code>R</code></li>
<li><code>merge_tree</code>: allows evolution evidences to be merged and compared</li>
<li><code>fortify</code> methods: convert tree objects into tidy data frame</li>
</ul>
<p></font></p>
<h2 id="tree-visualization-annotation">Tree visualization &amp; annotation</h2>
<p><font size='5'></p>
<ul>
<li>parsing tree as a collection of nodes allows grammar of graphics to be supported</li>
<li><code>geom_tree</code>: extends <code>ggplot2</code> to support tree structure</li>
<li>several layers and functions for tree annotation</li>
<li>supports annotating phylogenetic trees with user’s own data</li>
</ul>
<p></font></p>
<h2 id="tree-manipulation-6">Tree manipulation</h2>
<p><font size='5'></p>
<ul>
<li>helper functions for tree manipulation, make it possible to explore the tree visually</li>
</ul>
<p></font></p>
</div>
</section>
<section id="in-code-we-trust" class="slide level1">
<h1>IN CODE WE TRUST</h1>
<h2 id="questions-and-comments">Questions and comments?</h2>
<ul>
<li><a href="http://guangchuangyu.github.io/ggtree" class="uri">http://guangchuangyu.github.io/ggtree</a></li>
<li><a href="https://github.com/GuangchuangYu/ggtree" class="uri">https://github.com/GuangchuangYu/ggtree</a></li>
<li><a href="http://www.bioconductor.org/packages/ggtree" class="uri">http://www.bioconductor.org/packages/ggtree</a></li>
<li><a href="https://groups.google.com/forum/#!forum/bioc-ggtree" class="uri">https://groups.google.com/forum/#!forum/bioc-ggtree</a></li>
</ul>
</section>

  <!--
  To hide progress bar from entire presentation
  just remove “progress” element.
  -->
  <!-- <div class="progress"></div> -->
  <script src="index_files/rmdshower/shower/node_modules/shower-core/shower.min.js"></script>
  <!-- Copyright © 2015 Yours Truly, Famous Inc. -->
  <!-- Photos by John Carey, fiftyfootshadows.net -->

    <script>renderMathInElement(document.body);</script>
  
  
</body>
</html>
